### ISSUE TITLE
Wrong language parameter for not translated pages

### ISSUE THREAD

**Issue by timohund** [TYPO3-Solr/ext-solr#531]
I have two languages on a page.

default/0 = german
1 = english

If a page is not translated to english just the english url is indexed by the german solr core and there is no entry
in the english core.
e.g. the german core has a doc with

```
<str name="url">/en/blog/</str>
```

The english core just returns

```
<result name="response" numFound="0" start="0"/>
```

If I translate the page and content to english the german url occurs in the german core and the english doc is added to the english core.

German core

```
<str name="url">/blog/</str>
```

English core

```
<str name="url">/en/blog/</str>
```

The language and solr TypoScript setup:

```
config{
  linkVars = L
  sys_language_uid = 0
  language = de
  locale_all = de_DE.utf8
  htmlTag_langKey = de
}

# tx_solr configuration
plugin.tx_solr {
  solr.host = user:password@ip
  solr.path = /solr/core_de/
  suggest = 1
  search.results.resultsHighlighting = 1
  search.sorting = 1
  search.spellchecking = 1
  search.faceting = 1

  # disable jQuery
  javascriptFiles.library >
}

[globalVar = GP:L = 1]
config.sys_language_uid = 1
config.language = en
config.locale_all = en_US.utf8
config.htmlTag_langKey = en

plugin.tx_solr.solr.path = /solr/core_/
[global]
```

Environment:
TYPO3 6.0.4
tx_solr 2.8.2 and git master.
PHP 5.3.10

https://forge.typo3.org/issues/47111

**Comment by timohund** [comment]
**1**
Updated by Ingo Renner over 3 years ago
Comment Edit

Hi Jan,

Thanks for reporting the issue, but please do not set the target version yourself. It's meant for release planning only.

**2**
Updated by Ingo Renner over 3 years ago

    Priority changed from Must have to Should have
    TYPO3 Version changed from 6.0 to 4.5

Comment Edit

Also, did you have a chance to look into this already, trying to locate the bug in PHP?
**3**
Updated by Jan Kiesewetter over 3 years ago
Comment Edit

Hi Ingo,
sorry for the wrong configured issue.

I haven't took a look inside the code.
I first thought of a configuration fault, because this bug has not been reported yet.
**4**
Updated by Jan Kiesewetter over 3 years ago
Comment Edit

The devlog shows that both docs where send to the same core with the same id.
First the default language, than the secound.
So the doc with the default language is overridden.
**5**
Updated by Ingo Renner over 3 years ago

    Status changed from New to Needs Feedback
    Assigned To set to Jan Kiesewetter

Comment Edit

It actually might be a configuration issue, I have seen that behavior before, indexing both languages into one core, overwriting each other... unfortunately I can't remember the reason though.

How did you configure language fallback?

P.S. no offense taken on the target version ;)
**6**
Updated by Jan Kiesewetter over 3 years ago
Comment Edit

The sys_language_mode was unconfigured to use the default behaviour.
(http://docs.typo3.org/typo3cms/TyposcriptReference/Setup/Config/Index.html)

If i set:

```
config.sys_language_mode = content_fallback
```

the default language doc is send to the default core and the fallback doc to the english core.

that solves the problem for my installation.

But on multi domain and multi language installations with one domain that just uses the default language it is still a problem because the languages are global, e.g.:

```
domain1.tld
german english

domain2.tld
german

Website Language
english on pid 0
```

than domain2.tld will send 2 docs with the same id to the same core.

An option would be to respect page TSconfig mod.SHARED.disableLanguages to solve this problem in such a setup.
http://docs.typo3.org/typo3cms/TSconfigReference/PageTsconfig/Mod/Index.html

**7**
Updated by Ingo Renner over 3 years ago

    Status changed from Needs Feedback to New
    Assigned To deleted (Jan Kiesewetter)
    Target version deleted (2.8.x)

Comment Edit

moving this out for now

**8**
Updated by Lorenz Ulrich over 2 years ago
Comment Edit

For the record: Post 6 by Jan describes a real use-case and therefore this is still an issue.

**Comment by dmitryd** [comment]
I hit this problem too. Responsible code is here: https://github.com/TYPO3-Solr/ext-solr/blob/master/Classes/Typo3PageIndexer.php#L170

Basicaly, the wrong core is chosen. The page is not translated, therefore `$this->page->sys_language_uid` is always 0.

This has nasty side effects. Suppose you have "/en/" => sys_language_uid=0 and "/de/" => sys_language_uid=1, and try to index /de/page/, which is not translated. It will go to the core for the default language. As a result, when you switch to /de/, you cannot find anything for that page (even in English) but when you search inside /en/, you will get a link to /de/.

The solution would be to select the core based on `$_GET['L']` because this is how conditions to select the core are made in TypoScript according to the documentation.

**Comment by lorenzulrich** [comment]
IMO, this is solved in #1418

**Comment by timohund** [comment]
@lorenzulrich Thanks, i think so too. I'll close the issue
